ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Initiator.java

```diff
@@ -171,6 +176,7 @@ public void run() {
 
           for (CompactionInfo ci : potentials) {
             try {
+              Database d = resolveDatabaseAndCache(ci);
```

Comment:
i don't think we should be doing this. We do not support suppress of auto-compaction on db level, just table.

Reply:
Only the pool can be set on DB level. The code is not looking for any other properties, like `no_auto_compaction`. The DB level pool assignment was a specific requirement from Janos, to make it easier to assign a whole DB to some dedicated pool.

Reply:
Only by introducing a new dedicated method in the ObjectStore. I'm not sure this worth that.